Neos.Flow.Log/Classes/DefaultLogger.php

```diff
@@ -123,17 +165,31 @@ public function removeBackend(Backend\BackendInterface $backend)
      */
     public function log($message, $severity = LOG_INFO, $additionalData = null, $packageKey = null, $className = null, $methodName = null)
     {
-        if ($packageKey === null) {
-            $backtrace = debug_backtrace(false);
-            $className = isset($backtrace[1]['class']) ? $backtrace[1]['class'] : null;
-            $methodName = isset($backtrace[1]['function']) ? $backtrace[1]['function'] : null;
-            $explodedClassName = explode('\\', $className);
-            // FIXME: This is not really the package key:
-            $packageKey = isset($explodedClassName[1]) ? $explodedClassName[1] : '';
+        if ($this->psrLogger === null) {
+            $this->psrLogger = $this->createDefaultPsrLogger();
         }
-        foreach ($this->backends as $backend) {
-            $backend->append($message, $severity, $additionalData, $packageKey, $className, $methodName);
+
+        $psrLogLevel = self::LOGLEVEL_MAPPING[$severity] ?? LOG_INFO;
+
+        if ($additionalData === null) {
+            $additionalData = [];
+        }
+
+        $additionalData = is_array($additionalData) ? $additionalData : ['additionalData' => $additionalData];
```

Comment:
So… if I pass an array having the key `className` and pass the `$className` argument, the latter will override my additional data entry. I'd say `$additionalData = ['additionalData' => $additionalData]` unconditionally.

Reply:
Again part of backwards compatibility. I would agree to what you say, but that will mean some legacy places will probably not log the className anymore.